client/components/data/query-terms/index.jsx

```diff
@@ -1,68 +1,39 @@
 import isShallowEqual from '@wordpress/is-shallow-equal';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestSiteTerms } from 'calypso/state/terms/actions';
 import { isRequestingTermsForQuery } from 'calypso/state/terms/selectors';
 
-class QueryTerms extends Component {
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
+const request = ( siteId, taxonomy, query ) => ( dispatch, getState ) => {
+	if ( siteId && ! isRequestingTermsForQuery( getState(), siteId, taxonomy, query ) ) {
+		dispatch( requestSiteTerms( siteId, taxonomy, query ) );
 	}
+};
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if (
-			this.props.siteId === nextProps.siteId &&
-			this.props.taxonomy === nextProps.taxonomy &&
-			isShallowEqual( this.props.query, nextProps.query )
-		) {
-			return;
-		}
-
-		this.request( nextProps );
+function useMemoized( obj ) {
+	const memoObj = useRef();
+	if ( ! isShallowEqual( obj, memoObj.current ) ) {
+		memoObj.current = obj;
 	}
+	return memoObj;
+}
 
-	request( props ) {
-		if ( props.requesting || ! props.siteId ) {
-			return;
-		}
-
-		props.requestSiteTerms( props.siteId, props.taxonomy, props.query );
-	}
+function QueryTerms( { siteId, taxonomy, query = {} } ) {
+	const dispatch = useDispatch();
+	const memoizedQuery = useMemoized( query ).current;
```

Comment:
Let's return `memoObj.current` from the hook. We want to return the value itself, not a ref. That the hook is using a ref internally is a private implementation detail.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/56572/commits/5c3ca83c865b3185fc0c3f17cb9bd757600f7952 👍🏻